num2words/lang_AR.py

```diff
@@ -0,0 +1,121 @@
+
+class Num2Word_AR(lang_EU.Num2Word_EU):
+    def set_high_numwords(self, high):
+        max = 3 + 3*len(high)
+        for word, n in zip(high, range(max, 3, -3)):
+            self.cards[10**n] = word + "illion"
+
+    def setup(self):
+        self.negword = "سالب "
+        self.pointword = "فاصلة"
+        self.errmsg_nornum = "Only numbers may be converted to words."
+        self.exclude_title = ["و", "فاصلة", "سالب"]
+
+        self.mid_numwords = [(1000000, "مليون"),(1000, "ألف"), (100, "مئة"),
+                             (90, "تسعين"), (80, "ثمانين"), (70, "سبعين"),
+                             (60, "ستين"), (50, "خمسين"), (40, "أربعين"),
+                             (30, "ثلاثين")]
+        self.low_numwords = ["عشرين", "تسعة عشر", "ثمانية عشر", "سبعة عشر",
+                             "ستة عشر", "خمسة عشر", "أربعة عشر", "ثلاثة عشر",
+                             "اثناعشر", "أحد عشر", "عشرة", "تسعة", "ثمانية",
+                             "سبعة", "ستة", "خمسة", "أربعة", "ثلاثة", "اثنين",
+                             "واحد", "صفر"]
+        self.ords = { "واحد"    : "أول",
+                      "اثنين"    : "ثاني",
+                      "ثلاثة"  : "ثالث",
+                      "أربعة": "رابع",
+                      "خمسة"   : "خامس",
+                      "ثمانية"  : "ثامن",
+                      "تسعة"   : "تاسع",
+                      "اثناعشر" : "ثاني عشر" }
+
+
+    def merge(self, lpair, rpair):
+        ltext, lnum = lpair
+        rtext, rnum = rpair
+        if lnum == 1 and rnum < 100:
+            return (rtext, rnum)
+        elif 100 > lnum > rnum :
+            return ("%s و%s"%(rtext, ltext), rnum + lnum)
+        elif lnum >= 100 > rnum:
+            return ("%s و %s"%(ltext, rtext), lnum + rnum)
+        elif rnum > lnum:
+            if lnum == 1 and rnum in [100, 1000, 1000000]:
+                return ("%s"%(rtext), rnum * lnum)
+            if lnum == 2 and rnum == 100:
+                return ("مئتين", rnum * lnum)
+            if lnum == 2 and rnum in [100, 1000]:
+                return ("%sين"%(rtext), rnum * lnum)
+            return ("%s %s"%(ltext, rtext), lnum * rnum)
+        return ("%s، %s"%(ltext, rtext), lnum + rnum)
+
+
+    def to_ordinal(self, value):
+        self.verify_ordinal(value)
+        outwords = self.to_cardinal(value).split(" ")
+        lastwords = outwords[-1].split("-")
+        lastword = lastwords[-1].lower()
+        try:
+            lastword = self.ords[lastword]
+        except KeyError:
+            lastword += ""
+        lastwords[-1] = self.title(lastword)
+        outwords[-1] = "،".join(lastwords)
+        return " ".join(outwords)
+
+
+    def to_ordinal_num(self, value):
+        self.verify_ordinal(value)
+        return "%s%s"%(value, self.to_ordinal(value)[-2:])
+
+
+    def to_year(self, val, longval=True):
+        if not (val//100)%10:
+            return self.to_cardinal(val)
+        return self.to_splitnum(val, hightxt="hundred", jointxt="and",
```

Comment:
I don't know any Arabic, but I don't think that you use "hundred" and "and", could you check this, please?